### PR TITLE
[Triton] Add auto filling config.pbtxt from template files.

### PR DESCRIPTION
**Pull request by yuekaizhang**
This PR:
  1. Add wenetspeech transducer stateless 5 support (using tokens.txt rather than bpe.model)
  2. Add a bash script build.sh to auto generate config files.
  3. Update export_onnx.py to save the model specific information for build.sh (encoder_dim, layers, left context etc.)
  4. Create config.pbtxt.template files, which could be modified via build.sh to generate config.pbtxt 
  5. Change submodules name: conformer_transducer -> transducer, greedy_search -> scorer (Next PR Would unify the fast beam search and greedy search repo in scorer.)
  6. Optimize model_repo_offline/scorer/1/model.py to reduce H2D, D2H copy 

@csukuangfj  @wd929  Would you mind checking it, thanks.

### CODE REVIEW

File: triton/scripts/onnx_triton_utils.py [hunk 1]
```diff
@@ -126,7 +126,16 @@ def forward(
...
+        embed = self.encoder_embed(x)
+
+        # cut off 1 frame on each size of embed as they see the padding
+        # value which causes a training and decoding mismatch.
+        embed = embed[:, 1:-1, :]
+
+        embed, pos_enc = self.encoder_pos(embed, self.left_context)
+        embed = embed.permute(1, 0, 2)  # (B, T, F) -> (T, B, F)
+
+        src_key_padding_mask = make_pad_mask(lengths, embed.size(0))
```
Comment (yuekaizhang) [inline triton/scripts/onnx_triton_utils.py:138]: (Btw, this is where I have to make this PR https://github.com/k2-fsa/icefall/pull/670 for dynamic batch with paded audios. )

File: triton/README.md [hunk 2]
```diff
@@ -69,26 +69,27 @@ cp <bpe_model_path> <jit_model_dir>
 
 Streaming Model Export:
 ```bash
-cp $SHERPA/triton/scripts/*onnx*.py $ICEFALL_DIR/egs/librispeech/ASR/pruned_stateless_transducer3/
+cp $SHERPA/triton/scripts/*onnx*.py $ICEFALL_DIR/egs/wenetspeech/ASR/pruned_stateless_transducer5/
```
Comment (csukuangfj) [inline triton/README.md:72]: Looks like you are using `SHERPA_SRC` before but here you are using `SHERPA`.

**Comment by wd929** [comment]
We'd better add some details about differences between Chinese and English models in README? e.g. the differences of the tokenizers?

File: triton/README.md (hunk 2 above)
Comment (yuekaizhang) [reply triton/README.md]: Thanks, fixed it.

**Comment by yuekaizhang** [comment]
> We'd better add some details about differences between Chinese and English models in README? e.g. the differences of the tokenizers?

Yes. I think the differences mainly come from pruned_transducer_stateless3 and pruned_transducer_stateless5. For different languages, the tokenizer_file is the only difference. I think maybe we could split this README.md into mutliple files: e.g. sherpa/triton/docs/client.md, sherpa/triton/docs/server.md, sherpa/triton/docs/benchmark.md as the current README.md is too large to read.

What's your opinion about location of this docs directory, sherpa/triton/docs/*.md or sherpa/docs/source/triton/*.rst, @csukuangfj ?

File: triton/model_repo_offline/scorer/1/model.py [hunk 3]
```diff
@@ -139,25 +160,20 @@ def execute(self, requests):
             # Extract the output tensors from the inference response.
             decoder_out = pb_utils.get_output_tensor_by_name(inference_response,
                                                             'decoder_out')
-            decoder_out = torch.utils.dlpack.from_dlpack(decoder_out.to_dlpack()).cpu().numpy()
-            #decoder_out = decoder_out.as_numpy()
+            decoder_out = from_dlpack(decoder_out.to_dlpack())
```
[the remark this answers is not in the document]
Comment (yuekaizhang) [reply triton/model_repo_offline/scorer/1/model.py:163]: Actually not, https://github.com/triton-inference-server/python_backend/blob/main/README.md#interoperability-and-gpu-support, it implements zero-copy tensors between triton and torch.

**Comment by csukuangfj** [comment]
We have
- https://github.com/k2-fsa/sherpa/tree/master/docs/source/ncnn
- https://github.com/k2-fsa/sherpa/tree/master/docs/source/onnx

I suggest that you can add `docs/source/triton`.
Please remember to also update
https://github.com/k2-fsa/sherpa/blob/master/docs/source/index.rst

**Comment by csukuangfj** [comment]
Please refer to
https://github.com/k2-fsa/sherpa/tree/master/docs
for how to build the documentation. You can visualize it locally.

**Comment by yuekaizhang** [comment]
> Please refer to https://github.com/k2-fsa/sherpa/tree/master/docs for how to build the documentation. You can visualize it locally.

I add the triton docs. Would you mind reviewing the updates?

**Comment by csukuangfj** [comment]
> > Please refer to https://github.com/k2-fsa/sherpa/tree/master/docs for how to build the documentation. You can visualize it locally.
> 
> I add the triton docs. Would you mind reviewing the updates? @csukuangfj

Thanks! Let me have a look.

**Comment by csukuangfj** [comment]
Thanks! Looks great to me.

**Comment by csukuangfj** [comment]
By the way, the doc is available at
https://k2-fsa.github.io/sherpa/triton/overview.html

File: docs/source/triton/perf/index.rst [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+.. code-block:: bash
+
+  # Offline ASR Test with grpc 
+  perf_analyzer -m transducer -b 1 -a -p 20000 --concurrency-range 100:200:50 -i gRPC --input-data=offline_input.json  -u localhost:8001
+
+  # Streaming ASR Test with grpc
+  perf_analyzer -m transducer -b 1 -a -p 20000 --concurrency-range 100:200:50 -i gRPC --input-data=online_input.json  -u localhost:8001 --streaming
+
+
+.. literalinclude:: ./log/offline_perf.txt
```
Comment (csukuangfj) [inline docs/source/triton/perf/index.rst:42]: @yuekaizhang 
Could you please commit this file to the git repo?

Comment (yuekaizhang) [reply docs/source/triton/perf/index.rst:42]: Sure, I will do it today.

**Pull request merged**